lib/workers/global/config/parse/file.ts

```diff
@@ -79,21 +79,22 @@ export async function getConfig(env: NodeJS.ProcessEnv): Promise<AllConfig> {
     logger.debug('No config file found on disk - skipping');
   }
 
-  await deleteNonDefaultConfig(env); // Try deletion only if RENOVATE_CONFIG_FILE is specified
+  await deleteNonDefaultConfig(env, !!config.deleteConfigFile); // Try deletion only if RENOVATE_CONFIG_FILE is specified
```

Comment:
this is a problem. We can't set `RENOVATE_DELETE_CONFIG_FILE` from env, because we don't pass it to the file `getConfig`. so the config value isn't yet available and causes a breaking change.
also cli flags are not yet available, so wer probably need to move the delete code to a later stage.

@rarkins WDYT?

Reply:
@nabeelsaabna can we move this to a later stage and still achieve the goal?

Reply:
In that case, we could move it to the `config/parse/index` file. We will have the env vars, and flags available there.